### PR TITLE
20156 - BCOL Admin Access - update ministries account GL

### PR DESCRIPTION
**Pull request by Jxio**
*Issue #:*
https://github.com/bcgov/entity/issues/20156

*Description of changes:*
1. GL codes list collapse at Staff Dashboard
image
2. User with bcol_admin_staff role can edit GL info
image
3. Convert AccountPaymentMethods.vue, GLPaymentForm.vue and GLCodesListView.vue to composition-api

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-web/src/components/auth/account-settings/payment/AccountPaymentMethods.vue [hunk 1]
```diff
@@ -103,269 +104,281 @@
...
 
-  private setSelectedPayment (payment) {
-    this.errorMessage = ''
-    this.selectedPaymentMethod = payment.selectedPaymentMethod
-    this.isBtnSaved = (this.isBtnSaved && !payment.isTouched) || false
-    this.paymentMethodChanged = payment.isTouched || false
-  }
+    const errorDialog = ref<InstanceType<typeof ModalDialog>>()
+
+    const currentOrganization = computed(() => orgStore.currentOrganization)
```
Comment (seeker25) [inline auth-web/src/components/auth/account-settings/payment/AccountPaymentMethods.vue:147]: I see this repeated alot.. might be time to create a composable to destruct into these below?

`const { currentOrganization, currentOrgAAddress, currentMembership... } = useOrgComposable()`

File: auth-web/src/components/auth/account-settings/payment/AccountPaymentMethods.vue [hunk 2]
```diff
@@ -103,269 +104,281 @@
...
+      state.paymentMethodChanged = payment.isTouched || false
     }
 
-    return disableSaveBtn
-  }
+    const isDisableSaveBtn = computed(() => {
+      let disableSaveBtn = false
+      if (state.isBtnSaved) {
+        disableSaveBtn = false
+      } else if ((state.selectedPaymentMethod === PaymentTypes.PAD && !state.padValid) ||
```
Comment (seeker25) [inline auth-web/src/components/auth/account-settings/payment/AccountPaymentMethods.vue:165]: These conditionals should probably be a function?

File: auth-web/src/components/auth/account-settings/payment/AccountPaymentMethods.vue [hunk 3]
```diff
@@ -103,269 +104,281 @@
...
-      this.isTOSandAcknowledgeCompleted = orgPayments.padTosAcceptedBy !== null || false
-      this.selectedPaymentMethod = this.currentOrgPaymentType || ''
+      if (isPaymentViewAllowed.value) {
+        state.savedOrganizationType =
+        ((currentOrganization.value?.orgType === Account.PREMIUM) &&
+          !currentOrganization.value?.bcolAccountId && currentOrganization.value?.accessType !== AccessType.GOVM)
+          ? Account.UNLINKED_PREMIUM : currentOrganization.value.orgType
+        state.selectedPaymentMethod = ''
+        const orgPayments: OrgPaymentDetails = await orgStore.getOrgPayments()
+        // TODO : revisit  if need
```
Comment (seeker25) [inline auth-web/src/components/auth/account-settings/payment/AccountPaymentMethods.vue:230]: Gun the TODO?

File: auth-web/src/components/auth/account-settings/payment/AccountPaymentMethods.vue [hunk 4]
```diff
@@ -103,269 +104,281 @@
...
-        },
-        bcOnlineCredential: this.bcolInfo
-      }
-    } else {
-      isValid = true
-      createRequestBody = {
-        paymentInfo: {
-          paymentMethod: this.selectedPaymentMethod
+
+      if (isValid) {
```
Comment (seeker25) [inline auth-web/src/components/auth/account-settings/payment/AccountPaymentMethods.vue:310]: Worth putting this in it's own function?

File: auth-web/src/components/auth/common/PaymentMethods.vue [hunk 5]
```diff
@@ -310,18 +315,22 @@ export default defineComponent({
       const paymentMethods = []
       if (props.currentOrgType) {
         const paymentTypes = paymentsPerAccountType[props.currentOrgType]
-        paymentTypes.forEach((paymentType) => {
-          if (PAYMENT_METHODS[paymentType]) {
-            paymentMethods.push(PAYMENT_METHODS[paymentType])
-          }
-        })
+        if (paymentTypes) {
```
Comment (seeker25) [inline auth-web/src/components/auth/common/PaymentMethods.vue:318]: (paymentTypes || []).forEach() might be easier

File: auth-web/src/components/auth/account-settings/payment/AccountPaymentMethods.vue [hunk 6]
```diff
@@ -103,269 +104,281 @@
...
+import { computed, defineComponent, onMounted, reactive, ref, toRefs, watch } from '@vue/composition-api'
 import { BcolProfile } from '@/models/bcol'
-import { KCUserProfile } from 'sbc-common-components/src/models/KCUserProfile'
 import ModalDialog from '@/components/auth/common/ModalDialog.vue'
 import PaymentMethods from '@/components/auth/common/PaymentMethods.vue'
 import { useOrgStore } from '@/stores/org'
 import { useUserStore } from '@/stores/user'
 
-@Component({
+export default defineComponent({
```
Comment (seeker25) [inline auth-web/src/components/auth/account-settings/payment/AccountPaymentMethods.vue:115]: Thank you for doing this

File: auth-web/src/components/auth/account-settings/payment/AccountPaymentMethods.vue (hunk 2 above)
Comment (seeker25) [reply auth-web/src/components/auth/account-settings/payment/AccountPaymentMethods.vue:165]: personally I'd break these down into payment types.. 

`isPadSelectedAndInvalid`
`isEftSelectedAndInvalid`
`isEjvSelectedAndInvalid`
`paymentMethodNotChangedAndEjvSelected`

**Review by seeker25 — COMMENTED** [review]
small comments

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2864-gv8y7kxe.web.app

File: auth-web/src/components/auth/account-settings/payment/AccountPaymentMethods.vue (hunk 1 above)
Comment (Jxio) [reply auth-web/src/components/auth/account-settings/payment/AccountPaymentMethods.vue]: added

File: auth-web/src/components/auth/account-settings/payment/AccountPaymentMethods.vue (hunk 2 above)
Comment (Jxio) [reply auth-web/src/components/auth/account-settings/payment/AccountPaymentMethods.vue]: fixed

File: auth-web/src/components/auth/account-settings/payment/AccountPaymentMethods.vue (hunk 3 above)
Comment (Jxio) [reply auth-web/src/components/auth/account-settings/payment/AccountPaymentMethods.vue]: removed

File: auth-web/src/components/auth/account-settings/payment/AccountPaymentMethods.vue (hunk 4 above)
Comment (Jxio) [reply auth-web/src/components/auth/account-settings/payment/AccountPaymentMethods.vue:336]: added

File: auth-web/src/components/auth/common/PaymentMethods.vue (hunk 5 above)
Comment (Jxio) [reply auth-web/src/components/auth/common/PaymentMethods.vue]: update to: paymentTypes?.forEach()

**Review by ochiu — APPROVED** [review]
Thanks, looks good. Just the one issue we discussed about the form changes after one save doesn't reset the save button

File: auth-web/src/components/auth/account-settings/payment/AccountPaymentMethods.vue [hunk 7]
```diff
@@ -45,7 +47,6 @@
     <v-divider class="my-10" />
     <div class="form__btns d-flex">
       <v-btn
-        v-can:CHANGE_PAYMENT_METHOD.disable
```
Comment (Jxio) [inline auth-web/src/components/auth/account-settings/payment/AccountPaymentMethods.vue:48]: fix for: "the form changes after one save doesn't reset the save button"
disable functionality should handle by  `:disabled="isDisableSaveBtn"`

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2864-gv8y7kxe.web.app

**Pull request merged**